lib/unison-util-relation/benchmarks/relation/Main.hs

```diff
@@ -1,27 +1,55 @@
-{- ORMOLU_DISABLE -} -- Remove this when the file is ready to be auto-formatted
 module Main where
-import Criterion.Main
-import qualified Unison.Util.Relation as R
-import System.Random
+
 import Control.Monad
+import qualified Data.Set as Set
+import System.Random
+import Test.Tasty.Bench
+import Unison.Prelude
+import Unison.Util.Relation (Relation)
+import qualified Unison.Util.Relation as R
+
+main :: IO ()
+main =
+  defaultMain
+    [ env (genRelations @Char @Char 10000 20) \rs ->
+        bgroup
+          "Relation"
+          [ bench "difference" $ nf (uncurry R.difference) rs,
+            bench "intersection" $ nf (uncurry R.intersection) rs,
+            bench "union" $ nf (uncurry R.union) rs
+          ],
+      env (genRelation @Char @Char 10000 2) \r ->
+        env (genSet @Char 100) \s ->
+          bgroup
+            "Relation"
+            [ bgroup
+                "subtractDom"
+                [ bench "old implementation" (nf (oldSubtractDom s) r),
+                  bench "new implementation" (nf (R.subtractDom s) r)
+                ]
+            ]
+    ]
+
+oldSubtractDom :: (Ord a, Ord b) => Set a -> Relation a b -> Relation a b
```

Comment:
Did you intend to keep this around after testing it?

Reply:
I guess it could be deleted now, yeah. I left it in 'cause it doesn't seem too harmful.